### PR TITLE
animated set

### PR DESCRIPTION
**Pull request by adrianpricop**

### CODE REVIEW

File: test/CardRender.swift [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+//  test
+//
+//  Created by apple on 07/06/2018.
+//  Copyright © 2018 wolfpack. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class cardRender
```
Comment (emil622) [inline test/CardRender.swift]: Classes must have capitalised naming, rename to "CardRender"

File: test/CardRender.swift [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+{
+    public func getSimbol(for card: Card) -> String
+    {
```
Comment (emil622) [inline test/CardRender.swift]: I recommend adding opening braces on the same line with the block/method body. We have this convention on all other projects.

File: test/CardRender.swift [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        case .diamond:
+            return "■"
+        case .oval:
+            return "●"
+        case .squiggle:
+            return "▲"
+        }
+        
+    }
+    public func getColor(for card: Card) -> UIColor
```
Comment (emil622) [inline test/CardRender.swift]: Separate methods with an empty line

File: test/ContainerView.swift [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+//  Copyright © 2018 wolfpack. All rights reserved.
+//
+
+import UIKit
+
+protocol DidFinishAnimating {
+    func FinishedAnimating()
+}
+
+class ContainerView: UIView{
```
Comment (emil622) [inline test/ContainerView.swift:15]: Add an empty space before {

File: test/ContainerView.swift [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+            }
+        }
+        didSet
+        {
+            if selectedCards.count != 0
+            {
+                if inPlayCards.count == buttons.count && inPlayCards.count < 13
+                {
+                    removeCards(for: true)
+                }else if inPlayCards.count < buttons.count
```
Comment (emil622) [inline test/ContainerView.swift]: Add an empty space before 'else'

File: test/ContainerView.swift [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+
+        }
+    }
+    
+    var buttons = [DrawButton]()
+    
+    var grid = Grid(layout: Grid.Layout.aspectRatio(8/5))
+    
+    func DidFinishAnimating() {
+
```
Comment (emil622) [inline test/ContainerView.swift]: Remove these empty lines

File: test/ContainerView.swift [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+            delegate?.FinishedAnimating()
+
+    }
+    
+    private var centerRect: CGRect
+    {
+        get
+        {
+            return CGRect(x: bounds.size.width * 0.025, y: bounds.size.height * 0.025, width: bounds.size.width * 0.95, height: bounds.size.height * 0.95)
```
Comment (emil622) [inline test/ContainerView.swift]: split CGRect constructor on multiple lines for better readability

File: test/ContainerView.swift [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+        for button in buttons
+        {
+            button.removeFromSuperview()
+        }
+        grid.cellCount = 0
+        buttons.removeAll()
+        inPlayCards.removeAll()
+    }
+    
+//    func shuffle()
```
Comment (emil622) [inline test/ContainerView.swift]: Remove commented code

File: test/SetGame.swift [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+                    inPlayCards[index].isSelected = false
+                    
+                }else
+                {
+                    inPlayCards[index].isSelected = true
+                    selectedCards[index] = inPlayCards[index]
+                }
+            }
+
+//            if selectedCards.count == 3
```
Comment (emil622) [inline test/SetGame.swift]: Remove commented code

**Review by emil622 — CHANGES_REQUESTED** [review]

File: test/ButtonExtension.swift [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+//
+//  Created by apple on 05/06/2018.
+//  Copyright © 2018 wolfpack. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+extension UIButton
+{
```
Comment (BogdanPintilei) [inline test/ButtonExtension.swift:13]: As a general convention when creating a new class, extension, struct etc. you shoud move the bracket on the declaration line with an empty space between and a empty line after

File: test/ButtonExtension.swift [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+        self.layer.borderWidth = 2
+        self.layer.borderColor = #colorLiteral(red: 1, green: 0.1491314173, blue: 0, alpha: 1)
+    }
+    
+    func noSelect()
+    {
+        self.layer.borderWidth = 0
+        self.layer.cornerRadius = 10
+    }
+}
```
Comment (BogdanPintilei) [inline test/ButtonExtension.swift]: Also as a general convention you should a single  empty line before the last bracket.

File: test/Card.swift [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+//  Created by apple on 05/06/2018.
+//  Copyright © 2018 wolfpack. All rights reserved.
+//
+
+import Foundation
+
+
+struct Card: Hashable {
+    
+
```
Comment (BogdanPintilei) [inline test/Card.swift]: Only one single line

File: test/Card.swift [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+    {
+        return identifier.hashValue
+    }
+    
+    static func == (lhs: Card ,rhs: Card) ->Bool
+    {
+        return lhs.identifier == rhs.identifier
+    }
+    
+//    static var identifierFactori = 0
```
Comment (BogdanPintilei) [inline test/Card.swift]: delete unused commented out code

File: test/ContainerView.swift [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+            }
+
+        }
+    }
+    
+    var buttons = [DrawButton]()
+    
+    var grid = Grid(layout: Grid.Layout.aspectRatio(8/5))
+    
+    func DidFinishAnimating() {
```
Comment (BogdanPintilei) [inline test/ContainerView.swift]: begin functions with a small letter

File: test/ContainerView.swift [hunk 15]
```diff
@@ -0,0 +1,298 @@
...
+                    removeCards(for: false)
+                }
+            }else if inPlayCards.count == oldValue.count
+            {
+                updateCards()
+            }
+        }
+    }
+    
+    var buttonDeckFrame = CGRect()
```
Comment (BogdanPintilei) [inline test/ContainerView.swift]: Group all the variables at the begining of the class.

File: test/IntExtension.swift [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+//  test
+//
+//  Created by apple on 05/06/2018.
+//  Copyright © 2018 wolfpack. All rights reserved.
+//
+
+import Foundation
+
+extension Int
+{
```
Comment (BogdanPintilei) [inline test/IntExtension.swift:12]: Refactor brackets.

File: test/SetGame.swift [hunk 17]
```diff
@@ -0,0 +1,288 @@
...
+{
+    var inPlayCards = [Card]()
+    
+    var deck = [Card]()
+    
+    var score = 0
+    
+    var selectedCards = [Int:Card]()
+    
+    var isSetCards = [Int:Card]()
```
Comment (BogdanPintilei) [inline test/SetGame.swift]: Delete empty spaces between variable declarations

File: test/SetGame.swift [hunk 18]
```diff
@@ -0,0 +1,288 @@
...
+    {
+        var identifier = 0
+        for color in Card.CardColor.allValues
+        {
+            for simbol in Card.CardSimbol.allValues
+            {
+                for number in Card.CardNumber.allValues
+                {
+                    for shading in Card.CardShading.allValues
+                    {
```
Comment (BogdanPintilei) [inline test/SetGame.swift]: IF you have time you could replace the for in for logic, and use something like the mapping with the map function

File: test/SetGame.swift [hunk 19]
```diff
@@ -0,0 +1,288 @@
...
+                        let card = Card(matching: false, isSelected: false, identifier: identifier, cardColor: color, cardSimbol: simbol, cardNumber: number, cardShading: shading)
```
Comment (BogdanPintilei) [inline test/SetGame.swift]: separeate the attributes from the Card initialization on single lines

File: test/ViewController.swift [hunk 20]
```diff
@@ -0,0 +1,178 @@
...
+//  Copyright © 2018 wolfpack. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController, DidFinishAnimating
+{
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        game.cardGenerator()
```
Comment (BogdanPintilei) [inline test/ViewController.swift:15]: create a multiple methods for all commands in the viewDidLoad and call them in viewDidLoad

File: test/ViewController.swift [hunk 21]
```diff
@@ -0,0 +1,178 @@
...
+        drawThreeCardsButton.isEnabled = false
+        showSetButton.isEnabled = false
+        newGameButton.isEnabled = false
+    }
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        cardsContainerView.buttonDeckFrame = CGRect(x: drawThreeCardsButton.frame.origin.x ,
+                                                y: drawThreeCardsButton.frame.origin.y - drawThreeCardsButton.frame.height - 8 ,
+                                                width: drawThreeCardsButton.frame.width,
+                                                height: drawThreeCardsButton.frame.height)
```
Comment (BogdanPintilei) [inline test/ViewController.swift:33]: The same as in viewDidLoad

File: test/ViewController.swift [hunk 22]
```diff
@@ -0,0 +1,178 @@
...
+        newGameButton.isEnabled = false
+    }
+    @IBOutlet var shuffleCards: UIRotationGestureRecognizer!
+    @IBAction func shuffleCards(_ sender: Any) {
+//        game.shuffleCards()
+//        updateViewFromModel()
+//        cardsContainerView.inPlayCards = game.inPlayCards
+        
+    }
+    @IBOutlet weak var cardsContainerView: ContainerView!
```
Comment (BogdanPintilei) [inline test/ViewController.swift]: Delete empty lines between outlets

File: test/ViewController.swift [hunk 23]
```diff
@@ -0,0 +1,178 @@
...
+    func updateViewFromModel()
+    {
+        let score = game.score
+        scoreBord.text = "score: \(score)"
+        
+
+    }
+    
+    
+}
```
Comment (BogdanPintilei) [inline test/ViewController.swift]: Refactor the class with all the conventios proposed until now

**Review by BogdanPintilei — CHANGES_REQUESTED** [review]

**Pull request merged**